apps/aega/src/aega_meta_tx.erl

```diff
@@ -92,7 +92,7 @@ gas(#ga_meta_tx{gas = Gas}) ->
 
 -spec gas_limit(tx(), non_neg_integer()) -> amount().
 gas_limit(#ga_meta_tx{gas = Gas, tx = InnerTx}, Height) ->
-    aetx:gas_limit(aetx_sign:tx(InnerTx), Height) + Gas.
+    aetx:inner_gas_limit(aetx_sign:tx(InnerTx), Height) + Gas.
```

Comment:
Not introduced by this PR, though it seems to me that the `aetx:gas_limit` for `ga_meta_tx`, that [recurses using `CB:gas_limit`](https://github.com/aeternity/aeternity/pull/2428/files#diff-b7fc65d08fbfee442d57ae86b1b2576cR259), violates the semantics assumed by `aext_pool:check_candidate`:
> %% In case 0 is returned, the tx will not be included in the micro block
> %% candidate by the mempool.

in which the inner tx may be invalid though included. Or is that code path never exercised hence that semantics can be made tighter (enforcing never error)?

Reply:
I have not followed yet what is impact of not complying with this semantics in micro block candidate building - in case this code path (error) is exercised.

Reply:
`aetx:gas_limit` will never return 0 for a `ga_meta_tx` there is nothing that can go wrong at this level for this transaction. If the inner transaction has a bad oracle TTL or is an offchain_channel_tx the MetaTx can still be included on the chain with the effects of the authentication...

Reply:
It wouldn't be wrong to deny such a Tx, but accepting it is also OK.